src/ext/service.rs

```diff
@@ -109,23 +127,23 @@ pub struct Sources<'a> {
     /// All `Source` trees available for this repository.
     sources: &'a sources::Sources,
     /// Do we need to iterate over our `context` field?
-    context: Option<(String, dc::Context)>,
+    context: Option<(String, Option<String>, dc::Context)>,
     /// Libraries
     libs: vec::IntoIter<(String, String)>,
 }
 
 impl<'a> Iterator for Sources<'a> {
-    type Item = Result<(String, &'a Source)>;
+    type Item = Result<(String, Option<String>, &'a Source)>;
 
     fn next(&mut self) -> Option<Self::Item> {
         // Check for a `dc::Context` using `take`, which moves data out of
         // an `Option` value and leaves `None` in its place,
         // simulataneously updating our internal state and keeping the
         // borrow checker happy.
-        if let Some((container_path, context)) = self.context.take() {
-            if let Some(source) = self.sources.find_by_context(&context) {
+        if let Some((container_path, repo_subdir, context)) = self.context.take() {
+            if let Some(source) = self.sources.find_by_alias(&context.human_alias().unwrap()) {
```

Comment:
As per the change lower down, sources are stored without the subdirectory part, so the context of a `Service` and a `Source` might not be identical any more. So we need to look it up by its alias, which will still match.